clustering/jgroups/extension/src/main/java/org/jboss/as/clustering/jgroups/subsystem/ProtocolResourceDefinition.java

```diff
@@ -201,6 +207,31 @@ public void registerAttributes(ManagementResourceRegistration registration) {
         for (AttributeDefinition attr : ATTRIBUTES) {
             registration.registerReadWriteAttribute(attr, null, writeHandler);
         }
+
+        OperationStepHandler typeReadHandler = new OperationStepHandler() {
+            @Override
+            public void execute(OperationContext context, ModelNode operation) throws OperationFailedException {
+                context.getResult().set(context.getCurrentAddressValue());
+            }
+        };
+        OperationStepHandler typeWriteHandler = new OperationStepHandler() {
+            @Override
+            public void execute(OperationContext context, ModelNode operation) throws OperationFailedException {
+                // Issue remove operation for old resource address
+                // Followed by add operation for new resource address
+                String type = Operations.getAttributeValue(operation).asString();
+                PathAddress address = context.getCurrentAddress();
+                ModelNode removeOperation = Util.createRemoveOperation(address);
+                context.addStep(removeOperation, context.getResourceRegistration().getOperationHandler(PathAddress.EMPTY_ADDRESS, ModelDescriptionConstants.REMOVE), context.getCurrentStage());
+                PathAddress newAddress = address.getParent().append(PathElement.pathElement(address.getLastElement().getKey(), type));
+                ModelNode addOperation = Util.createAddOperation(newAddress);
+                for (Property property : context.readResource(PathAddress.EMPTY_ADDRESS, false).getModel().asPropertyList()) {
+                    addOperation.get(property.getName()).set(property.getValue());
+                }
+                context.addStep(addOperation, context.getRootResourceRegistration().getOperationHandler(newAddress, ModelDescriptionConstants.ADD), context.getCurrentStage());
```

Comment:
I recommend you include the "addFirst" param to the two context.addStep calls, with a value of "true". This will ensure that if this op is part of a composite op, the remove + add will execute before the next step in the composite, maintaining the logical sequence of the composite. The way it is now, these new steps will happen after all the steps in the composite.

Note though that you'll have to switch the order in which you add the steps, i.e. first add and then remove. The "addFirst=true" call is effectively putting the steps at the front of the queue, rather than at the end of a queue.